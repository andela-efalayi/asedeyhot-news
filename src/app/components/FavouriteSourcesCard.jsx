import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardText } from 'material-ui/Card';

const FavouriteSourcesCard = ({ favouriteSources }) => {
  if (Object.keys(favouriteSources).length === 0) {
    return (
      <div>
        <h4>You have no saved sources.</h4>
      </div>
    );
  }
  return (
    <div>
      <Card className="col-xs-12">
        <CardText>
          {Object.entries(favouriteSources)
          .map(([key, value]) => (
            <p key={key}>{key}: <a href={value}>{value}</a></p>
          ))}
        </CardText>
      </Card>
    </div>
  );
};

FavouriteSourcesCard.propTypes = {
  favouriteSources: PropTypes.object
};

FavouriteSourcesCard.defaultProps = {
  favouriteSources: null
};

export default FavouriteSourcesCard;
