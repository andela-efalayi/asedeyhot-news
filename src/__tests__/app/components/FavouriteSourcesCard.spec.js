import React from 'react';
import { shallow } from 'enzyme';
import FavouriteSourcesCard from
'../../../app/components/FavouriteSourcesCard.jsx';

const mockData = {
  id: 'abc-news-au',
  name: 'ABC News (AU)',
  description: "Australia's most trusted source of local, national news",
  url: 'http://www.abc.net.au/news',
  category: 'general',
  language: 'en',
  country: 'au',
  urlsToLogos: {
    small: '',
    medium: '',
    large: ''
  },
  sortBysAvailable: ['top']
};

describe('Favourites Component', () => {
  it('should render Favourites', () => {
    const wrapper = shallow(
        <FavouriteSourcesCard favouriteSources={mockData}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
