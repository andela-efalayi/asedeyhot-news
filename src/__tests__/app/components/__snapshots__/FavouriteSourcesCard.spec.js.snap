// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites Component should render Favourites 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Card
        className="col-xs-12"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
    >
        <CardText>
            <p>
                id
                : 
                <a
                    href="abc-news-au"
                >
                    abc-news-au
                </a>
            </p>
            <p>
                name
                : 
                <a
                    href="ABC News (AU)"
                >
                    ABC News (AU)
                </a>
            </p>
            <p>
                description
                : 
                <a
                    href="Australia's most trusted source of local, national news"
                >
                    Australia's most trusted source of local, national news
                </a>
            </p>
            <p>
                url
                : 
                <a
                    href="http://www.abc.net.au/news"
                >
                    http://www.abc.net.au/news
                </a>
            </p>
            <p>
                category
                : 
                <a
                    href="general"
                >
                    general
                </a>
            </p>
            <p>
                language
                : 
                <a
                    href="en"
                >
                    en
                </a>
            </p>
            <p>
                country
                : 
                <a
                    href="au"
                >
                    au
                </a>
            </p>
            <p>
                urlsToLogos
                : 
                <a
                    href={
                        Object {
                            "large": "",
                            "medium": "",
                            "small": "",
                          }
                    }
                >
                    Object {
                        "large": "",
                        "medium": "",
                        "small": "",
                      }
                </a>
            </p>
            <p>
                sortBysAvailable
                : 
                <a
                    href={
                        Array [
                            "top",
                          ]
                    }
                >
                    top
                </a>
            </p>
        </CardText>
    </Card>
</div>,
  "nodes": Array [
    <div>
      <Card
            className="col-xs-12"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
      >
            <CardText>
                  <p>
                        id
                        : 
                        <a
                              href="abc-news-au"
                        >
                              abc-news-au
                        </a>
                  </p>
                  <p>
                        name
                        : 
                        <a
                              href="ABC News (AU)"
                        >
                              ABC News (AU)
                        </a>
                  </p>
                  <p>
                        description
                        : 
                        <a
                              href="Australia's most trusted source of local, national news"
                        >
                              Australia's most trusted source of local, national news
                        </a>
                  </p>
                  <p>
                        url
                        : 
                        <a
                              href="http://www.abc.net.au/news"
                        >
                              http://www.abc.net.au/news
                        </a>
                  </p>
                  <p>
                        category
                        : 
                        <a
                              href="general"
                        >
                              general
                        </a>
                  </p>
                  <p>
                        language
                        : 
                        <a
                              href="en"
                        >
                              en
                        </a>
                  </p>
                  <p>
                        country
                        : 
                        <a
                              href="au"
                        >
                              au
                        </a>
                  </p>
                  <p>
                        urlsToLogos
                        : 
                        <a
                              href={
                                    Object {
                                          "large": "",
                                          "medium": "",
                                          "small": "",
                                        }
                              }
                        >
                              Object {
                                    "large": "",
                                    "medium": "",
                                    "small": "",
                                  }
                        </a>
                  </p>
                  <p>
                        sortBysAvailable
                        : 
                        <a
                              href={
                                    Array [
                                          "top",
                                        ]
                              }
                        >
                              top
                        </a>
                  </p>
            </CardText>
      </Card>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <FavouriteSourcesCard
        favouriteSources={
                Object {
                        "category": "general",
                        "country": "au",
                        "description": "Australia's most trusted source of local, national news",
                        "id": "abc-news-au",
                        "language": "en",
                        "name": "ABC News (AU)",
                        "sortBysAvailable": Array [
                          "top",
                        ],
                        "url": "http://www.abc.net.au/news",
                        "urlsToLogos": Object {
                          "large": "",
                          "medium": "",
                          "small": "",
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "favouriteSources": Object {
            "category": "general",
            "country": "au",
            "description": "Australia's most trusted source of local, national news",
            "id": "abc-news-au",
            "language": "en",
            "name": "ABC News (AU)",
            "sortBysAvailable": Array [
              "top",
            ],
            "url": "http://www.abc.net.au/news",
            "urlsToLogos": Object {
              "large": "",
              "medium": "",
              "small": "",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Card
                    className="col-xs-12"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <CardText>
                              <p>
                                        id
                                        : 
                                        <a
                                                  href="abc-news-au"
                                        >
                                                  abc-news-au
                                        </a>
                              </p>
                              <p>
                                        name
                                        : 
                                        <a
                                                  href="ABC News (AU)"
                                        >
                                                  ABC News (AU)
                                        </a>
                              </p>
                              <p>
                                        description
                                        : 
                                        <a
                                                  href="Australia's most trusted source of local, national news"
                                        >
                                                  Australia's most trusted source of local, national news
                                        </a>
                              </p>
                              <p>
                                        url
                                        : 
                                        <a
                                                  href="http://www.abc.net.au/news"
                                        >
                                                  http://www.abc.net.au/news
                                        </a>
                              </p>
                              <p>
                                        category
                                        : 
                                        <a
                                                  href="general"
                                        >
                                                  general
                                        </a>
                              </p>
                              <p>
                                        language
                                        : 
                                        <a
                                                  href="en"
                                        >
                                                  en
                                        </a>
                              </p>
                              <p>
                                        country
                                        : 
                                        <a
                                                  href="au"
                                        >
                                                  au
                                        </a>
                              </p>
                              <p>
                                        urlsToLogos
                                        : 
                                        <a
                                                  href={
                                                            Object {
                                                                      "large": "",
                                                                      "medium": "",
                                                                      "small": "",
                                                                    }
                                                  }
                                        >
                                                  Object {
                                                            "large": "",
                                                            "medium": "",
                                                            "small": "",
                                                          }
                                        </a>
                              </p>
                              <p>
                                        sortBysAvailable
                                        : 
                                        <a
                                                  href={
                                                            Array [
                                                                      "top",
                                                                    ]
                                                  }
                                        >
                                                  top
                                        </a>
                              </p>
                    </CardText>
          </Card>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Card
                    className="col-xs-12"
                    expandable={false}
                    expanded={null}
                    initiallyExpanded={false}
          >
                    <CardText>
                              <p>
                                        id
                                        : 
                                        <a
                                                  href="abc-news-au"
                                        >
                                                  abc-news-au
                                        </a>
                              </p>
                              <p>
                                        name
                                        : 
                                        <a
                                                  href="ABC News (AU)"
                                        >
                                                  ABC News (AU)
                                        </a>
                              </p>
                              <p>
                                        description
                                        : 
                                        <a
                                                  href="Australia's most trusted source of local, national news"
                                        >
                                                  Australia's most trusted source of local, national news
                                        </a>
                              </p>
                              <p>
                                        url
                                        : 
                                        <a
                                                  href="http://www.abc.net.au/news"
                                        >
                                                  http://www.abc.net.au/news
                                        </a>
                              </p>
                              <p>
                                        category
                                        : 
                                        <a
                                                  href="general"
                                        >
                                                  general
                                        </a>
                              </p>
                              <p>
                                        language
                                        : 
                                        <a
                                                  href="en"
                                        >
                                                  en
                                        </a>
                              </p>
                              <p>
                                        country
                                        : 
                                        <a
                                                  href="au"
                                        >
                                                  au
                                        </a>
                              </p>
                              <p>
                                        urlsToLogos
                                        : 
                                        <a
                                                  href={
                                                            Object {
                                                                      "large": "",
                                                                      "medium": "",
                                                                      "small": "",
                                                                    }
                                                  }
                                        >
                                                  Object {
                                                            "large": "",
                                                            "medium": "",
                                                            "small": "",
                                                          }
                                        </a>
                              </p>
                              <p>
                                        sortBysAvailable
                                        : 
                                        <a
                                                  href={
                                                            Array [
                                                                      "top",
                                                                    ]
                                                  }
                                        >
                                                  top
                                        </a>
                              </p>
                    </CardText>
          </Card>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FavouriteSourcesCard
    favouriteSources={
        Object {
            "category": "general",
            "country": "au",
            "description": "Australia's most trusted source of local, national news",
            "id": "abc-news-au",
            "language": "en",
            "name": "ABC News (AU)",
            "sortBysAvailable": Array [
              "top",
            ],
            "url": "http://www.abc.net.au/news",
            "urlsToLogos": Object {
              "large": "",
              "medium": "",
              "small": "",
            },
          }
    }
/>,
}
`;
